README.md

```diff
@@ -240,22 +240,527 @@ For complete documentation of the `mp-build` tool see the [MP Build Tool Documen
+```
+Inside the `AdapterKind` tag are `ResourceKinds/ResourceKind` tags, and we can search 
+for the one that represents the database resource kind. Once we have found it we can see
+that it has two identifiers, one for the adapter instance ID, and one for the database 
+name. 
+```xml
+   <ResourceKinds>
+      <!-- ... -->
+      <ResourceKind key="mysql_database" nameKey="64" >
+          <ResourceIdentifier dispOrder="1" key="adapter_instance_id" length="" nameKey="37" required="true" type="string" identType="1" enum="false" default=""> </ResourceIdentifier>
+          <ResourceIdentifier dispOrder="2" key="database_name" length="" nameKey="65" required="true" type="string" identType="1" enum="false" default=""> </ResourceIdentifier>
+```
+In order to attach a metric to these objects, we will need all identifiers that have an
+`identType=1`. In this case, those are `adapter_instance_id` and `database_name`. This
+means that the combination of those two fields uniquely identify the object among all
+of the `mysql_database` objects in the `MySQLAdapter` adapter.
+
+Unfortunately, getting the `adapter_instance_id` is tricky. However, if we look in Aria 
+Ops itself, we can see that each database's name has the format 
+`mysql_host/mysql_database`, which should be unique (even if VMware Aria Operations 
+isn't using it for determining uniqueness). We can construct the name, ask the SuiteAPI 
+to give us all `MySQLAdapter` `mysql_database` objects with that name, and attach
+metrics to the result.
+
+First, we need to establish a connection to MySQL. We can do this in the same way as in
+test connect. In many cases creating a function for connecting that is called from both 
+`test` and `collect` is worthwhile. Then we can query the list of databases, and 
+construct a list of database names that may be present:
+
+```python
+        # Get the list of databases on this instance
+        cursor = connection.cursor()
+        cursor.execute("SHOW databases")
+        database_names = [f"{hostname}/{database[0]}" for database in cursor]
+        cursor.close()
+```
+
+We then query the SuiteAPI for `mysql_database` objects from the `MySQLAdapter` 
+adapter, with the names we computed. The queries that `query_for_resources` accepts
+are documented in the SuiteAPI documentation, and can search on many different types
+of metadata about a resource. After that we add the returned objects to our `result`
+and, also add them to a dictionary so we can access them easily later.
+
+```python
+        # Get the list of objects from the SuiteAPI that represent the MySQL
+        # databases that are on this instance, and add any we find to the result
+        databases = {}  # dict of database Objects by name for easy access
+        with adapter_instance.suite_api_client as suite_api:
+            dbs = suite_api.query_for_resources(
+                query={
+                    "adapterKind": [MYSQL_ADAPTER_TYPE],
```

Comment:
Need to instruct user to set these constants (MYSQL_ADAPTER_TYPE and MYSQL_DATABASE_OBJECT_TYPE)

Reply:
or, alternatively, just replace the constants with a placeholder string